presto-hive/src/main/java/com/facebook/presto/hive/metastore/HivePrivilegeInfo.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.metastore;
+
+import com.facebook.presto.spi.security.Privilege;
+import com.google.common.collect.ImmutableSet;
+import org.apache.hadoop.hive.metastore.api.PrivilegeGrantInfo;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.hive.metastore.HivePrivilegeInfo.HivePrivilege.DELETE;
+import static com.facebook.presto.hive.metastore.HivePrivilegeInfo.HivePrivilege.INSERT;
+import static com.facebook.presto.hive.metastore.HivePrivilegeInfo.HivePrivilege.OWNERSHIP;
+import static com.facebook.presto.hive.metastore.HivePrivilegeInfo.HivePrivilege.SELECT;
+import static com.facebook.presto.hive.metastore.HivePrivilegeInfo.HivePrivilege.UPDATE;
+import static java.util.Locale.ENGLISH;
+
+public class HivePrivilegeInfo
+{
+    HivePrivilegeInfo(HivePrivilege hivePrivilege, boolean grantOption)
+    {
+        this.hivePrivilege = hivePrivilege;
+        this.grantOption = grantOption;
+    }
+
+    public HivePrivilege getHivePrivilege()
+    {
+        return hivePrivilege;
+    }
+
+    public void setHivePrivilege(HivePrivilege hivePrivilege)
+    {
+        this.hivePrivilege = hivePrivilege;
+    }
+
+    public boolean isGrantOption()
+    {
+        return grantOption;
+    }
+
+    public void setGrantOption(boolean grantOption)
+    {
+        this.grantOption = grantOption;
+    }
+
+    public enum HivePrivilege
+    {
+        SELECT, INSERT, UPDATE, DELETE, OWNERSHIP;
+    }
+
+    private HivePrivilegeInfo.HivePrivilege hivePrivilege;
```

Comment:
Re-arranged the members of the class. 

Regarding making the class immutable, I can think of one scenario where we may need to change the state: Suppose that a user 'alice' has been granted the INSERT privilege. Later, if we want to add the GRANT option for the INSERT privilege for this user, we will have to change the state of the existing HivePrivilege object (assuming that the implementation of GRANT will allow adding such a GRANT option on an existing privilege).